contracts/crowdsale/price/IncreasingPriceCrowdsale.sol

```diff
@@ -0,0 +1,39 @@
+pragma solidity ^0.4.18;
+
+import "../validation/TimedCrowdsale.sol";
+import "../../math/SafeMath.sol";
+
+
+contract IncreasingPriceCrowdsale is TimedCrowdsale {
+  using SafeMath for uint256;
+
+  uint256 public initialRate;
+  uint256 public finalRate;
+
+  /**
+   * @param _initialRate Number of tokens a buyer gets per wei at the start of the crowdsale
+   * @param _finalRate Number of tokens a buyer gets per wei at the end of the crowdsale
+   */
+  function IncreasingPriceCrowdsale(uint256 _initialRate, uint256 _finalRate) public {
+    require(_initialRate >= _finalRate);
+    require(_finalRate > 0);
+    initialRate = _initialRate;
+    finalRate = _finalRate;
+  }
+
+  /**
+   * @return The number of tokens a buyer gets per wei at a given time
+   */
+  function getCurrentRate() public view returns (uint256) {
+    uint256 elapsedTime = now - startTime;
```

Comment:
I suggest using `SafeMath` in here.

As this line is only protected against overflow way up in the hierarchy, another option is to make this a `private` function and explain in a comment the fact that this function is meant to be called only by this contract's `_getTokenAmount` after `_preValidatePurchase` in `buyTokens`, where the `TimedCrowdsale` rejects `now < startTime` purchases, but I think that's risky and this public function might be useful for dApps.